x-pack/plugins/siem/server/lib/detection_engine/signals/filter_events_with_list.test.ts

```diff
@@ -0,0 +1,160 @@
+    try {
+      const res = await filterEventsAgainstList({
+        logger: mockLogger,
+        listClient: ({
+          getListItemByValues: async () => ([] as unknown) as ListItemArraySchema,
+        } as unknown) as ListClient,
+        exceptionsList: [
+          {
+            field: 'source.ip',
+            values_operator: 'excluded',
+            values_type: 'list',
+            values: [
+              {
+                id: 'ci-badguys.txt',
+                name: 'unsupportedListPluginType',
+              },
+            ],
+          },
+        ],
+        eventSearchResult: repeatedSearchResultsWithSortId(4, 4, someGuids.slice(0, 3), [
+          '1.1.1.1',
+          '2.2.2.2',
+          '3.3.3.3',
+          '7.7.7.7',
+        ]),
+      });
+      expect(res.hits.hits.length).toEqual(4);
+    } catch (exc) {
+      expect(exc.message).toContain('Unsupported list type used, please use one of');
+    }
+  });
+
+  it('should respond with same list if no items match value list', async () => {
+    const res = await filterEventsAgainstList({
+      logger: mockLogger,
+      listClient: ({
+        getListItemByValues: async () => ([] as unknown) as ListItemArraySchema,
+      } as unknown) as ListClient,
+      exceptionsList: [
+        {
+          field: 'source.ip',
+          values_operator: 'excluded',
+          values_type: 'list',
+          values: [
+            {
+              id: 'ci-badguys.txt',
+              name: 'ip',
+            },
+          ],
+        },
+      ],
+      eventSearchResult: repeatedSearchResultsWithSortId(4, 4, someGuids.slice(0, 3)),
+    });
+    expect(res.hits.hits.length).toEqual(4);
+  });
+  it('should respond with less items in the list if some values match', async () => {
+    const res = await filterEventsAgainstList({
+      logger: mockLogger,
+      listClient: ({
+        getListItemByValues: async ({
+          value,
+          type,
+          listId,
+        }: {
+          type: string;
+          listId: string;
+          value: string[];
+        }) => {
+          expect(type).toEqual('ip');
+          expect(listId).toEqual('ci-badguys.txt');
+          const toReturn = value.slice(0, 2).map((item) => ({
+            value: item,
+          }));
+          return (toReturn as unknown) as ListItemArraySchema;
+        },
+      } as unknown) as ListClient,
+      exceptionsList: [
+        {
+          field: 'source.ip',
+          values_operator: 'excluded',
+          values_type: 'list',
+          values: [
+            {
+              id: 'ci-badguys.txt',
+              name: 'ip',
+            },
+          ],
+        },
+      ],
+      eventSearchResult: repeatedSearchResultsWithSortId(4, 4, someGuids.slice(0, 3), [
+        '1.1.1.1',
+        '2.2.2.2',
+        '3.3.3.3',
+        '7.7.7.7',
+      ]),
+    });
+    expect(res.hits.hits.length).toEqual(2);
+  });
+});
```

Comment:
For this one, rather than have the expects within the function call back which at a later time could end up looking like it is working but really that function is no longer being called, I would set it up like so below:

```ts
  it('should respond with less items in the list if some values match', async () => {
    let outerType = '';
    let outerListId = '';
    const res = await filterEventsAgainstList({
      logger: mockLogger,
      listClient: ({
        getListItemByValues: async ({
          value,
          type,
          listId,
        }: {
          type: string;
          listId: string;
          value: string[];
        }) => {
          outerType = type;
          outerListId = listId;
          return value.slice(0, 2).map((item) => ({
            value: item,
          }));
        },
      } as unknown) as ListClient,
      exceptionsList: [
        {
          field: 'source.ip',
          values_operator: 'excluded',
          values_type: 'list',
          values: [
            {
              id: 'ci-badguys.txt',
              name: 'ip',
            },
          ],
        },
      ],
      eventSearchResult: repeatedSearchResultsWithSortId(4, 4, someGuids.slice(0, 3), [
        '1.1.1.1',
        '2.2.2.2',
        '3.3.3.3',
        '7.7.7.7',
      ]),
    });
    expect(outerType).toEqual('ip');
    expect(outerListId).toEqual('ci-badguys.txt');
    expect(res.hits.hits.length).toEqual(2);
  });
```

Then you know that if suddenly the function is no longer called your test will fail. Alternatively you could setup a Jest spy and then inspect the arguments called on the jest spy.